developer/src/kmc-package/src/compiler/package-version-validation.ts

```diff
@@ -116,24 +129,57 @@ export class PackageVersionValidation {
       kmxFileData = this.callbacks.loadFile(filename);
     } catch(e) {
       this.callbacks.reportMessage(CompilerMessages.Error_FileCouldNotBeRead({filename, e}));
-      return false;
+      return null;
     }
     const kmxReader: KmxFileReader = new KmxFileReader();
     const kmx: KMX.KEYBOARD = kmxReader.read(kmxFileData);
     if(!kmx) {
       // The file couldn't be read, it might not be a .kmx file
       this.callbacks.reportMessage(CompilerMessages.Error_KeyboardFileNotValid({filename}));
-      return false;
+      return null;
     }
 
+    return kmx;
+  }
+
+  private getKeyboardVersionFromKmx(
+    kmx: KMX.KEYBOARD
+  ): string {
     const store = kmx.stores.find(store => store.dwSystemID == KMX.KMXFile.TSS_KEYBOARDVERSION);
+    return store ? store.dpString : null;
```

Comment:
Do you want to fall back to `DEFAULT_VERSION` instead of null?

Reply:
No, see the caller on l.56 which does additional processing if there is no keyboard version specified.